src/test/pybind/test_cephfs.py

```diff
@@ -431,3 +492,279 @@ def test_futimens():
+    assert_equal(st["gid"], gid)
+    cephfs.fchown(fd, 9999, 9999)
+    st = cephfs.statx(b'/file-fchown', libcephfs.CEPH_STATX_UID | libcephfs.CEPH_STATX_GID, 0)
+    assert_equal(st["uid"], 9999)
+    assert_equal(st["gid"], 9999)
+    cephfs.close(fd)
+    cephfs.unlink(b'/file-fchown')
+
+@with_setup(setup_test)
+def test_truncate():
+    fd = cephfs.open(b'/file-truncate', 'w', 0o755)
+    cephfs.write(fd, b"1111", 0)
+    cephfs.truncate(b'/file-truncate', 0)
+    stat = cephfs.fsync(fd, 0)
+    st = cephfs.statx(b'/file-truncate', libcephfs.CEPH_STATX_SIZE, 0)
+    assert_equal(st["size"], 0)
+    cephfs.close(fd)
+    cephfs.unlink(b'/file-truncate')
+
+@with_setup(setup_test)
+def test_ftruncate():
+    fd = cephfs.open(b'/file-ftruncate', 'w', 0o755)
+    cephfs.write(fd, b"1111", 0)
+    assert_raises(TypeError, cephfs.ftruncate, b'/file-ftruncate', 0)
+    cephfs.ftruncate(fd, 0)
+    stat = cephfs.fsync(fd, 0)
+    st = cephfs.fstat(fd)
+    assert_equal(st.st_size, 0)
+    cephfs.close(fd)
+    cephfs.unlink(b'/file-ftruncate')
+
+@with_setup(setup_test)
+def test_fallocate():
+    fd = cephfs.open(b'/file-fallocate', 'w', 0o755)
+    assert_raises(TypeError, cephfs.fallocate, b'/file-fallocate', 0, 10)
+    cephfs.fallocate(fd, 0, 10)
+    stat = cephfs.fsync(fd, 0)
+    st = cephfs.fstat(fd)
+    assert_equal(st.st_size, 10)
+    cephfs.close(fd)
+    cephfs.unlink(b'/file-fallocate')
+
+@with_setup(setup_test)
+def test_mknod():
+    mode = stat.S_IFIFO | stat.S_IRUSR | stat.S_IWUSR
+    cephfs.mknod(b'/file-fifo', mode)
+    st = cephfs.statx(b'/file-fifo', libcephfs.CEPH_STATX_MODE, 0)
+    assert_equal(st["mode"] & mode, mode)
+    cephfs.unlink(b'/file-fifo')
+
+@with_setup(setup_test)
+def test_lazyio():
+    fd = cephfs.open(b'/file-lazyio', 'w', 0o755)
+    assert_raises(TypeError, cephfs.lazyio, "fd", 1)
+    assert_raises(TypeError, cephfs.lazyio, fd, "1")
+    cephfs.lazyio(fd, 1)
+    cephfs.write(fd, b"1111", 0)
+    assert_raises(TypeError, cephfs.lazyio_propagate, "fd", 0, 4)
+    assert_raises(TypeError, cephfs.lazyio_propagate, fd, "0", 4)
+    assert_raises(TypeError, cephfs.lazyio_propagate, fd, 0, "4")
+    cephfs.lazyio_propagate(fd, 0, 4)
+    st = cephfs.fstat(fd)
+    assert_equal(st.st_size, 4)
+    cephfs.write(fd, b"2222", 4)
+    assert_raises(TypeError, cephfs.lazyio_synchronize, "fd", 0, 8)
+    assert_raises(TypeError, cephfs.lazyio_synchronize, fd, "0", 8)
+    assert_raises(TypeError, cephfs.lazyio_synchronize, fd, 0, "8")
+    cephfs.lazyio_synchronize(fd, 0, 8)
+    st = cephfs.fstat(fd)
+    assert_equal(st.st_size, 8)
+    cephfs.close(fd)
+    cephfs.unlink(b'/file-lazyio')
+
+@with_setup(setup_test)
+def test_replication():
+    fd = cephfs.open(b'/file-rep', 'w', 0o755)
+    assert_raises(TypeError, cephfs.get_file_replication, "fd")
+    cnt = cephfs.get_file_replication(fd)
+    assert_equal(cnt, 3)
+    cnt = cephfs.get_path_replication(b'/file-rep')
+    assert_equal(cnt, 3)
```

Comment:
Same as `test_get_pool`, replication factor is hardcoded which might led to a false positive.

Reply:
Done